docs/platform/tutorials/eventDetection/crashDetection.md

```diff
@@ -0,0 +1,277 @@
+
+		class QuixFunction:
+			def __init__(self, input_stream: StreamReader, output_stream: StreamWriter, loaded_model):
+				self.input_stream = input_stream
+				self.output_stream = output_stream
+				self.loaded_model = loaded_model
+
+			# Callback triggered for each new event.
+			def on_event_data_handler(self, data: EventData):
+				print(data.value)
+				# Here transform your data.
+				self.output_stream.events.write(data)
+
+			# Callback triggered for each new parameter data.
+			def on_pandas_frame_handler(self, df: pd.DataFrame):
+				
+				# if the expected column is in the dataframe
+				if "gForceX" in df:
+					# calc total g-force
+					df["gForceTotal"] = df["gForceX"].abs() +  df["gForceY"].abs() + df["gForceZ"].abs()
+					# predict 'shaking'
+					df["shaking"] = self.loaded_model.predict(df[["gForceZ", "gForceY", "gForceX", "gForceTotal"]])
+
+					# if 'shaking'
+					if df["shaking"].max() == 1: 
+						print("Crash detected.")
+
+						# write the event to the output stream
+						self.output_stream.events.add_timestamp_nanoseconds(df.iloc[0]["time"]) \
+							.add_value("crash", "Crash detected.") \
+							.write()
+		```
+	
+### Test again
+
+You can once again run the code in the development environment to test the functionality.
+
+1. Ensure that the data source you deployed earlier is running
+
+2. Click `Run` in the top right of the browser to run the event detection code
+
+3. If you chose to stream live data then gently shake your phone 
+
+4. If you went for the CSV data then wait for a crash event to be streamed from the data set or stop and start the service in another browser tab
+
+5. Observe the `Console` tab. You should see a message saying "Crash detected"
+
+4. On the `Messages` tab select `output : phone-out` from the first drop down
+
+5. Gently shake your phone, or wait for another crash event from the CSV data, and observe that crash events are streamed to the output topic. You can click these rows to investigate the event data e.g. 
+
+	```
+	[{
+		"Timestamp": 1674137774598972000,
+		"Tags": {},
+		"Id": "crash",
+		"Value": "Crash detected."
+	}]
+	```
+
+6. Stop the code
+
+!!! success
+	The crash detection service is working as expected and can now be deployed
+
+### Deploy crash detection
+
+Now that you have verified the service is working you can go ahead and deploy the service.
+
+1. Tag the code by clicking the `add tag` icon at the top of the code panel
+
+2. Enter a tag such as `v1`
+
+3. Now click the `Deploy` button near the top right of the code panel
+
+4. From the `Version tag` drop down, select the tag you created
+
+5. Click `Deploy`
+
+!!! success
+	You now have a data source and the crash detection service running in your workspace. 
+	
+	Next you’ll deploy a real-time UI to visualize the route being taken, the location of any crash events and also to see some of the sensor data.
+
+In the next part, deploy the demos '[Crash detection UI](crashDetectionUI.md)'
```

Comment:
In other tutorials you used a nice arrangement like:

```
[Deploy the first part of the solution by following step 1 :material-arrow-right-circle:{ align=right }](sentiment-demo-ui.md)
```

I think that format for the "next step" is quite nice and we should apply it consistently as we update tutorials. This comment also applies to other instances below.